### DISCUSSION TITLE
empty container warning

### DISCUSSION THREAD

**Discussion by prim** [microsoft/pyright#8159] [General]
```python
d = {}
l = []
```
```text
> pyright a5.py
0 errors, 0 warnings, 0 informations

> mypy a5.py
a5.py:1: error: Need type annotation for "d" (hint: "d: Dict[<type>, <type>] = ...")  [var-annotated]
a5.py:2: error: Need type annotation for "l" (hint: "l: List[<type>] = ...")  [var-annotated]
Found 2 errors in 1 file (checked 1 source file)
```

Is there any option for pyright to report empty containers?

**Comment by erictraut** [comment]
I'm not entirely sure what you mean by "report empty containers". Empty containers are often used in Python code, and they are typically not a problem from a typing standpoint. Often, a type checker can infer their type from a [bidirectional inference](https://microsoft.github.io/pyright/#/type-inference?id=bidirectional-type-inference-expected-types) context. In cases where pyright can't do this, it will infer the type `dict[Unknown, Unknown]` or `list[Unknown]`.

If you want to detect the use of a value with an unknown or partially-unknown type, you can use the `reportUnknownXXX` family of diagnostic checks. These are off by default in "standard" type checking mode, but they're on by default in "strict" mode.

Code sample in [pyright playground](https://pyright-play.net/?code=MQAgDgngTglg5gCwC4C4QGcmwMZILABQhAJiALwgDaAuiQHQCGYYApgHbEAUAjAJQigAKhFYgA9gDMQAIiasO0kDHTgGUJDAYAbLRBABXNgGs2YgO5tChIA)

```python
# pyright: strict

d = []
d.append(1) # Type of "append" is partially unknown
```
